artemis/reporting/blocklist.py

```diff
@@ -1,19 +1,22 @@
 import dataclasses
 import datetime
-from typing import List, Optional
+import ipaddress
+from typing import List, Optional, Union
 
 import yaml
 
-from artemis.domains import is_subdomain
+from artemis.domains import is_domain, is_subdomain
 from artemis.reporting.base.report import Report
 from artemis.reporting.base.report_type import ReportType
 
 
 @dataclasses.dataclass
 class BlocklistItem:
-    domain: str
-    until: Optional[datetime.datetime]
-    report_type: Optional[ReportType]
+    domain: Optional[str] = None
+    ip_range: Optional[Union[ipaddress.IPv4Network, ipaddress.IPv6Network]] = None
+    target_should_contain: Optional[str] = None
```

Comment:
I think, it would be better as a `List[str]`, as it could open up more robust filters on targets